File: backend/app/models/evento.py
from enum import Enum
from typing import Optional, List
from datetime import datetime, date, time
from pydantic import BaseModel, Field

# FASE 5 — TIPOS DE EVENTOS
class TipoEvento(str, Enum):
    INICIO_LABOR = "INICIO_LABOR"
    INTERRUPCION = "INTERRUPCION"
    REANUDACION = "REANUDACION"
    FIN_LABOR = "FIN_LABOR"

# FASE 6 — CATEGORÍAS OFICIALES DE IA
class CategoriaIA(str, Enum):
    PRODUCCION_ACTIVA = "Producción Activa"
    ALISTAMIENTO_PREPARACION = "Alistamiento y Preparación (Setup)"
    ESPERA_MATERIALES = "Espera de Materiales / Logística"
    FALLA_TECNICA = "Falla Técnica / Mantenimiento"
    CALIDAD_APROBACION = "Calidad y Aprobación"
    INSTRUCCIONES_COORDINACION = "Instrucciones / Coordinación"

# FASE 4 — MODELOS DE DATOS
class Operario(BaseModel):
    id: str
    nombre: str
    codigo: str
    rol: str

class OrdenProduccion(BaseModel):
    id: str
    codigo: str
    descripcion: str
    estado: str

class Maquina(BaseModel):
    id: str
    nombre: str
    tipo: str
    estado: str

class Evento(BaseModel):
    id: str
    fecha: str
    hora_inicio: str
    hora_fin: Optional[str] = None
    operario_id: str
    orden_id: str
    maquina_id: str
    tipo_evento: TipoEvento
    mensaje: Optional[str] = None
    categoria: Optional[CategoriaIA] = None
    descripcion: Optional[str] = None
    confianza_ia: Optional[float] = None
    duracion_minutos: Optional[int] = None
    estado: str = "PROCESADO"

# FASE 7 - Modelos para Endpoint de Clasificación IA
class MensajeClasificacion(BaseModel):
    mensaje: str

class ResultadoClasificacionIA(BaseModel):
    categoria: CategoriaIA
    descripcion: str
    confianza: float

File: app/src/main/java/com/superbrix/ia/data/model/Evento.kt
package com.superbrix.ia.data.model

object IACategorias {
    const val PRODUCCION_ACTIVA = "Producción Activa"
    const val SETUP = "Alistamiento y Preparación (Setup)"
    const val LOGISTICA_MATERIALES = "Espera de Materiales / Logística"
    const val FALLA_TECNICA = "Falla Técnica / Mantenimiento"
    const val CALIDAD = "Calidad y Aprobación"
    const val COORDINACION = "Instrucciones / Coordinación"
    const val SALUD_SEGURIDAD = "Salud y Seguridad (Accidentes)"
    const val OTROS = "Otros / Novedad Excepcional"

    val TODAS = listOf(
        PRODUCCION_ACTIVA,
        SETUP,
        LOGISTICA_MATERIALES,
        FALLA_TECNICA,
        CALIDAD,
        COORDINACION,
        SALUD_SEGURIDAD,
        OTROS
    )
}

data class Evento(
    val id: String,
    val fecha: String,
    val horaInicio: String,
    val horaFin: String? = null,
    val operario: String = "Sara",
    val ordenProduccion: String = "OP-001",
    val maquina: String = "Fresadora 01",
    val tipoEvento: String, // e.g. "Inicio de labor", "Interrupción"
    val descripcion: String,
    val categoria: String, // One of IACategorias
    val duracionMinutos: Int = 0,
    val duracionSegundos: Int = 0,
    val confianzaIA: String = "94%"
)
